test_conformance/images/samplerlessReads/test_loops.cpp

```diff
@@ -114,14 +113,57 @@ int test_image_set( cl_device_id device, cl_context context, cl_command_queue qu
         return TEST_SKIPPED_ITSELF;
     }
 
-    // This flag is only for querying the list of supported formats
-    // The flag for creating image will be set explicitly in test functions
-    cl_mem_flags flags = (gTestReadWrite)? CL_MEM_KERNEL_READ_AND_WRITE : CL_MEM_READ_ONLY;
+    cl_image_format *readOnlyFormats;
+    unsigned int numReadOnlyFormats;
 
-    if ( get_format_list( context, imageType, formatList, numFormats, flags ) )
+    if (get_format_list(context, imageType, readOnlyFormats, numReadOnlyFormats,
+                        CL_MEM_READ_ONLY))
         return -1;
 
-    filterFlags = new bool[ numFormats ];
+    if (gTestReadWrite)
+    {
+        cl_image_format *readWriteFormats;
+        unsigned int numReadWriteFormats;
+
+        if (get_format_list(context, imageType, readWriteFormats,
+                            numReadWriteFormats, CL_MEM_KERNEL_READ_AND_WRITE))
+            return -1;
+
+        numFormats = numReadOnlyFormats;
```

Comment:
Maybe this should be numReadOnlyFormats + numReadWriteFormats ?

Reply:
I agree that the original code wasn't safe there, though the expectation is that there will be at most numReadOnlyFormats matches in the following nested loops.  The patch has been updated to include a break after finding a match in the inner loop, guaranteeing at most one match per iteration of the outer loop, which should make this both safe and explicit.